src/broker/content-cache.c

```diff
@@ -605,6 +605,133 @@ static void content_store_request (flux_t *h, flux_msg_handler_t *mh,
         flux_log_error (h, "content store: flux_respond_error");
 }
 
+static void checkpoint_get_continuation (flux_future_t *f, void *arg)
+{
+    struct content_cache *cache = arg;
+    const flux_msg_t *msg = flux_future_aux_get (f, "msg");
+    const char *s;
+
+    assert (msg);
+
+    if (flux_rpc_get (f, &s) < 0)
+        goto error;
+
+    if (flux_respond (cache->h, msg, s) < 0) {
+        flux_log_error (cache->h, "%s: flux_respond", __FUNCTION__);
+        goto error;
+    }
+
+    flux_future_destroy (f);
+    return;
+
+error:
+    if (flux_respond_error (cache->h, msg, errno, NULL) < 0)
+        flux_log_error (cache->h, "flux_respond_error");
+    flux_future_destroy (f);
+}
+
+/* flux_free_f type */
+static void flux_msg_decref_wrapper (void *arg)
+{
+    const flux_msg_t *msg = arg;
+    flux_msg_decref (msg);
+}
```

Comment:
Just cast `flux_msg_decref()` to flux_free_f instread?

Reply:
ugh, get lots of warnings b/c the `flux_msg_incref/decref()` functions are const (which is probably why I wrote the wrapper to begin with and forgot)